packages/diagrams/frontend/sirius-components-diagrams/src/renderer/tools/useSingleClickTool.tsx

```diff
@@ -113,6 +123,41 @@ export const useSingleClickTool = (): UseSingleClickToolValue => {
     });
   };
 
+  const { getImpactAnalysisReport, loading: impactAnalysisLoading, impactAnalysisReport } = useInvokeImpactAnalysis();
+
+  const [state, setState] = useState<UseSingleClickToolState>({
+    currentTool: null,
+    onToolExecution: () => {},
+  });
```

Comment:
I had to use a state here to pass the information that is only known in `invokeSingleClickTool` to the `useEffect`.
Let me know if it needs to be modified, I have the feeling we shouldn't need a state but I don't know how to fix that.

Reply:
Yeah that seems quite odd to have a function in the state for example, this lifecycle is odd. I'll look into it.